core/config/configs.md

```diff
@@ -0,0 +1,152 @@
+do so, we will need to create our own custom configuration. 
+
+Our custom configuration will look like this:
+
+```js
+const makeTotalSupplyConfig = () => {
+
+  function* makePaymentTrait(_superPayment) {
+    yield harden({});
+  }
+
+  function* makePurseTrait(_superPurse) {
+    yield harden({});
+  }
+
+  function* makeMintTrait(_superMint, _issuer, _assay, mintKeeper) {
+    return yield harden({
+      getTotalSupply: () => mintKeeper.getTotalSupply(),
+    });
+  }
+
+  function* makeIssuerTrait(_superIssuer) {
+    yield harden({});
+  }
+
+  return harden({
+    makePaymentTrait,
+    makePurseTrait,
+    makeMintTrait,
+    makeIssuerTrait,
+    makeMintKeeper: makeTotalSupplyMintKeeper,
+    strategy: natStrategy,
+    makeMintTrait,
+  });
+};
+```
+
+In this custom configuration, we've done two things: we've added a
+method to `mint` called `getTotalSupply`, and we've changed the
+`mintKeeper` to a custom mintKeeper that keeps track of the total
+supply for us (more on this in separate documentation). 
+
+Let's take a look into how we are able to add new methods to mints,
+issuers, purses, and payments. 
+
+In `makePaymentTrait`, we take `superPayment` as a parameter. The
+`superPayment` is constructed in `issuers.js` and has all of the
+methods we are familiar with:
+
+```js
+const superPayment = harden({
+  getIssuer() {
+    return issuer;
+  },
+  getBalance() {
+    return paymentKeeper.getAmount(payment);
+  },
+  getName() {
+    return name;
+  },
+});
+```
+
+Our custom code, `makePaymentTrait`, returns an object with custom methods.
+These methods will be added to the `superPayment`. For this particular
+customization, we want to leave payments alone, so we will create a
+generator function that yields an empty object:
+
+```js
+function* makePaymentTrait(_superPayment) {
+  yield harden({});
+}
+```
+
+However, we do want to add an extra method to mints. So,
+`makeMintTrait` is defined as:
+
+```js
+function* makeMintTrait(_superMint, _issuer, _assay, mintKeeper) {
+  return yield harden({
+    getTotalSupply: () => mintKeeper.getTotalSupply(),
+  });
+}
+```
+
+Back in `issuers.js`, our custom methods will be combined with the
+"core" methods already present. Here's how that works for our custom
+mint methods:
+
+```js
+const makeMintTraitIter = makeMintTrait(coreMint, issuer, assay, mintKeeper);
+const mintTrait = makeMintTraitIter.next().value;
+const mint = harden({
+  ...mintTrait,
+  ...coreMint,
+});
+makeMintTraitIter.next(mint);
+```
+
+## The Trait Pattern
+
+As a general rule, we do not want the customization code to be able to
+override the methods already on the mints, issuers, purses or
+payments. It would hardly be a standard if the standard methods acted
+entirely differently! On the other hand, we do know that there is a
+genuine need to have slightly different purposes expressed. 
+
+This is why we've chosen to use the [Trait
+pattern](https://traitsjs.github.io/traits.js-website/), in which
```

Comment:
I think this is a [better link for traits](https://en.wikipedia.org/wiki/Trait_(computer_programming)). It describes the OO pattern rather than a JS package we're not using.